### ISSUE TITLE
pKeyFilter makes the form invalid

### ISSUE THREAD

**Issue by mubashshirul** [primefaces/primeng#5497]
I have a field 'Aadhar' which is not mandatory. But it makes the form invalid even if i do not fill the data in Aadhar field.

```
<input class="input-field" formControlName="Aadhar" placeholder="Aadhar ID"
               maxlength='12' pKeyFilter="pint" pInputText>
```

**Comment by adama357** [comment]
Yes, I think it would be better if pKeyFilter always considered empty input valid, since if we want a field to be required, we can just mark it required.

**Comment by JacobSiegle** [comment]
Is it fixed or am I setting something up incorrectly?

https://stackblitz.com/edit/github-hzv69y

**Comment by adama357** [comment]
Fixed: see #5047
